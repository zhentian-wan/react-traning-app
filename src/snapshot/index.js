import React from 'react'
export {makeSomeRequestAndGetSomeResponse, getFilteredStuff, MyComponent}

function makeSomeRequestAndGetSomeResponse() {
    return new Promise(resolve => {
        resolve({
                    name: 'Joe',
                    id: '3423490',
                    age: 32,
                    foo: 'bar',
                    email: '123@example.com',
                })
    })
}

function getFilteredStuff(filter) {
    return [
        'You shall not pass',
        'Vacuums have filters',
        'I will not be filtered out',
        'but I will',
    ].filter(i => i.includes(filter))
}

function MyComponent({name}) {
    return <div><h1>Hello</h1> there <strong>{`${name}!`}</strong></div>
}