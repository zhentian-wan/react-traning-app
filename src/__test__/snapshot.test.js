import React from 'react'
import TestUtils from 'react-dom/test-utils';
import { createRenderer as renderer }  from 'react-test-renderer/shallow';
import {makeSomeRequestAndGetSomeResponse, getFilteredStuff, MyComponent} from '../snapshot';

test('objects', async () => {
    const httpResponse = await makeSomeRequestAndGetSomeResponse();
    expect(httpResponse).toMatchSnapshot();
});


test('arrays', () => {
    const someFilteredList = getFilteredStuff('filter')
    expect(someFilteredList).toMatchSnapshot()
});

test('jsx', () => {
    const renderer = TestUtils.createRenderer();
    renderer.render(<MyComponent name="John" />);
    const component = renderer.getRenderOutput();
    expect(component).toMatchSnapshot()
});

test('jsx: example2', () => {
    const component = renderer(<MyComponent name="John" />); //shadow renderer
    expect(component).toMatchSnapshot()
})