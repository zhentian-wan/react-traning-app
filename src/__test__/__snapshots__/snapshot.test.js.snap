exports[`test arrays 1`] = `
Array [
  "Vacuums have filters",
  "I will not be filtered out",
]
`;

exports[`test jsx 1`] = `
<div>
  <h1>
    Hello
  </h1>
   there 
  <strong>
    John!
  </strong>
</div>
`;

exports[`test jsx: example2 1`] = `
ReactShallowRenderer {
  "_instance": null,
}
`;

exports[`test objects 1`] = `
Object {
  "age": 32,
  "email": "123@example.com",
  "foo": "bar",
  "id": "3423490",
  "name": "Joe",
}
`;
